app/lib/features/chat/pages/room_profile_page.dart

```diff
@@ -70,282 +74,314 @@ class RoomProfilePage extends ConsumerWidget {
-                icon: const Icon(
-                  Icons.bookmark_add_outlined,
-                  size: 20,
-                ),
-                onPressed: () {},
-              ),
-            ),
-          ),
-          Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 5),
-            child: IconButton.filled(
-              onPressed: () async {
-                final permaLink =
-                    await (await ref.read(chatProvider(roomId).future))
-                        .permalink();
-                Clipboard.setData(
-                  ClipboardData(
-                    text: permaLink,
-                  ),
-                );
-                EasyLoading.showToast(
-                  'Link to Room copied to clipboard',
-                );
+  Widget _actions(BuildContext context) {
+    final convoLoader = ref.watch(chatProvider(widget.roomId));
+    final myMembership = ref.watch(roomMembershipProvider(widget.roomId));
+
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        // Bookmark
+        convoLoader.when(
+          data: (conv) {
+            final isFav = conv.isFavorite();
+            return _actionItem(
+              context: context,
+              iconData: isFav ? Icons.bookmark : Icons.bookmark_border,
+              actionName: 'Bookmark',
+              onTap: () async {
+                await conv.setFavorite(!isFav);
               },
-              tooltip: 'Copy RoomLink',
+            );
+          },
+          error: (e, st) => Skeletonizer(
+            child: IconButton.filled(
               icon: const Icon(
-                Icons.copy_outlined,
+                Icons.bookmark_add_outlined,
                 size: 20,
               ),
+              onPressed: () {},
             ),
           ),
-        ],
-      ),
+          loading: () => const ActionItemSkeleton(
+            iconData: Icons.bookmark_add_outlined,
+            actionName: 'Bookmark',
+          ),
+        ),
+
+        // Invite
+        myMembership.when(
+          data: (membership) {
+            return _actionItem(
+              context: context,
+              iconData: Atlas.user_plus_thin,
+              actionName: 'Invite',
+              onTap: () {
+                membership!.canString('CanInvite')
+                    ? context.pushNamed(
+                        Routes.spaceInvite.name,
+                        pathParameters: {'spaceId': widget.roomId},
+                      )
+                    : customMsgSnackbar(
+                        context,
+                        'Not enough power level for invites, ask room administrator to change it',
+                      );
+              },
+            );
+          },
+          error: (e, st) => Text('Error loading tile due to $e'),
+          loading: () => const ActionItemSkeleton(
+            iconData: Atlas.user_plus_thin,
+            actionName: 'Invite',
+          ),
+        ),
```

Comment:
arguably showing the invite if you can't invite is a bit confusing. maybe we could show a disabled button with a tooltip stating the user can't invite to this room instead?

Reply:
Let me change colour of invite action button if user don't have permission.

Reply:
Done :)
<img width="366" alt="Screenshot 2024-03-08 at 10 03 51 PM" src="https://github.com/acterglobal/a3/assets/51526480/746e3ecf-77b2-4f2d-a226-5661b7f95759">